lib/internal/worker.js

```diff
@@ -92,7 +92,7 @@ let cwdCounter;
 
 const environmentData = new SafeMap();
 
-if (isMainThread) {
+if (isMainThread && typeof SharedArrayBuffer !== 'undefined') {
```

Comment:
If we don't add a test, a comment explaining  that SharedArrayBuffer can be disabled with the flag would be useful.